Appsilon/shiny.blueprint
# Make package production-ready

### Changes
Depends on PR #6; please review and merge it first. The goal of this PR is to make Filip's PoC "production-ready": fill in missing parts, improve consistency, define package code structure and technical solutions to be followed for the remaining components.

1. Add missing components (`AnchorButton`, `AnchorButton.shinyInput`, `EditableText`).
2. Organize and add examples for all components.
3. Define documentation structure (component groups using `@rdname`, params template using `@inherit`, titles and documentation links). Follow official Blueprint docs for order; add TODOs for missing components.
4. Refactor JS and ensure it builds cleanly.

Summary of all documentation pages along with components:
1. Button: `Button`, `Button.shinyInput`, `AnchorButton`, `AnchorButton.shinyInput`
3. Button group: `ButtonGroup`
5. Callout: `Callout`
6. Card: `Card`
7. Collapse: `Collapse`
8. Divider: `Divider`
9. Editable text: `EditableText`, `EditableText.shinyInput`
10. HTML elements: `H1`, `H2`, `H3`, `H4`, `H5`, `H6`, `Blockquote`, `Code`, `Label`, `Pre`, `OL`, `UL`, `HTMLTable`
11. Icon: `Icon`
12. Menu: `Menu`, `MenuItem`, `MenuDivider`
13. Navbar: `Navbar`, `NavbarGroup`, `NavbarHeading`, `NavbarDivider`
14. Non-ideal state: `NonIdealState`
15. Progress bar: `ProgressBar`
16. Spinner: `Spinner`
17. Switch: `Switch`, `Switch.shinyInput`
18. Popover: `Popover`

### How to test
1. Run `devtools::check()`. It should only complain about license.
2. Check that all examples work.
3. Review our [spreadsheet](https://docs.google.com/spreadsheets/d/10VDrw4p5J8mGrrhHaqwkWL7M6zkHR4y7jRqz079Zb3Q/edit#gid=0) & [Blueprint docs](https://blueprintjs.com/docs/) to ensure nothing is missing.
